src/canister_tests/src/framework.rs

```diff
@@ -513,23 +508,38 @@ pub fn time(env: &StateMachine) -> u64 {
         .as_nanos() as u64
 }
 
-pub fn verify_delegation(user_key: UserKey, signed_delegation: &SignedDelegation, root_key: &[u8]) {
-    // transform delegation into ic typed delegation so that we have access to the signature domain separator
-    // (via as_signed_bytes)
-    let delegation = Delegation::new(
-        signed_delegation.delegation.pubkey.clone().into_vec(),
-        Time::from_nanos_since_unix_epoch(signed_delegation.delegation.expiration),
+pub fn verify_delegation(
+    env: &StateMachine,
+    user_key: UserKey,
+    signed_delegation: &SignedDelegation,
+    root_key: &[u8],
+) {
+    const DOMAIN_SEPARATOR: &[u8] = b"ic-request-auth-delegation";
+
+    // calculate the delegation hash (which is the signed message)
+    let map = vec![
```

Comment:
nit: `map` is not very description and is almost exclusively used for `map: (A -> B) -> [A] -> [B]`